src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
@chinpei215 I consider this more a bugfix, because so far this kind of ISO date was wrongly not supported.
Are there any reasonable downsides of those "valid" date strings to be not turned down? Just saying "because so far wasnt allowed" is not a valid reason. Those already get exported like this for years, they must also be able get imported the same way.